apps/web-client/src/routes/orders/customers/+page.svelte

```diff
@@ -9,26 +9,48 @@
 
 	import { PageCenterDialog, defaultDialogConfig } from "$lib/components/Melt";
 	import CustomerOrderMetaForm from "$lib/forms/CustomerOrderMetaForm.svelte";
-	import { createCustomerOrderSchema } from "$lib/forms";
+	import { createCustomerOrderSchema, customerSearchSchema } from "$lib/forms";
 	import { appPath, appHash } from "$lib/paths";
 
 	import { base } from "$app/paths";
 
-	import { getCustomerDisplayIdSeq, upsertCustomer } from "$lib/db/cr-sqlite/customers";
+	import { getCustomerDisplayIdSeq, getCustomerOrderList, upsertCustomer } from "$lib/db/cr-sqlite/customers";
 	import { Page } from "$lib/controllers";
 
-	import type { Customer } from "$lib/db/cr-sqlite/types";
+	import type { Customer, CustomerOrderListItem } from "$lib/db/cr-sqlite/types";
 	import LL from "@librocco/shared/i18n-svelte";
 
 	import type { PageData } from "./$types";
 
+	import CustomerSearchForm from "$lib/forms/CustomerSearchForm.svelte";
+	import { writable } from "svelte/store";
+	import { createIntersectionObserver } from "$lib/actions";
+
 	export let data: PageData;
 
 	$: ({ customerOrders, plugins } = data);
 	$: db = data.dbCtx?.db;
 
+	let searchField: HTMLInputElement;
+	const search = writable("");
+
 	$: t = $LL.customer_orders_page;
+	const tableStore = writable<CustomerOrderListItem[]>(customerOrders);
+
+	const seeMore = () => {
+		maxResults += 10;
+	};
+	const scroll = createIntersectionObserver(seeMore);
+	let maxResults = 10;
 
+	$: filteredOrders = customerOrders
+		.filter(({ completed }) => completed === (orderFilterStatus === "completed"))
+		.filter(({ fullname, displayId }) => {
+			if (!$search) return true;
+			return fullname.toLowerCase().includes($search.toLowerCase()) || displayId.toLowerCase().includes($search.toLowerCase());
```

Comment:
One more thing I just noticed: here if `$search` has value ` doe, Jane` I'd like it to also match `Jane Doe`.

I suggest:
* putting code for this (we need to find a name for the function that checks is a given user should be included or not in the search results - `functionThatChecksIfAGivenUserShouldBeIncludedOrNotInTheSearchResults` has its merits, but we can probably find a better one) in a separate `.ts` file, and import it from here.
* add unit tests for common cases

[This conversation might help](https://kagi.com/assistant/eb466a68-ac3a-4c65-9826-da78c4b0fed2)

Reply:
Opening a new ticket for this